metricbeat/module/nats/_meta/Dockerfile

```diff
@@ -1,20 +1,17 @@
-ARG NATS_VERSION=2.0.4
+ARG NATS_VERSION=2.10.26
 FROM nats:$NATS_VERSION
 
-# build stage
-FROM golang:1.24.1 AS build-env
-RUN apt-get install git mercurial gcc
-RUN git clone https://github.com/nats-io/nats.go.git /nats-go
-RUN cd /nats-go/examples/nats-bench && git checkout tags/v1.10.0 && go build .
-
 # create an enhanced container with nc command available since nats is based
 # on scratch image making healthcheck impossible
 FROM alpine:latest
 COPY --from=0 / /opt/nats
-COPY --from=build-env /nats-go/examples/nats-bench/nats-bench /nats-bench
-COPY run.sh /run.sh
+COPY ./run.sh /run.sh
+COPY ./bench.sh /bench.sh
+RUN apk update && apk add --no-cache curl
+RUN curl -sf https://binaries.nats.dev/nats-io/natscli/nats@latest | sh
+
 # Expose client, management, and cluster ports
 EXPOSE 4222 8222 6222
 HEALTHCHECK --interval=1s --retries=10 CMD nc -w 1 0.0.0.0 8222 </dev/null
-# Run via the configuration file
+
```

Comment:
@JAndritsch integration tests are failing.

Looks the container used for integration tests never turns healthy:

> connection_integration_test.go:40: timeout waiting for services to be healthy

https://github.com/JAndritsch/beats/blob/3d1bded0f4b623c019b8c8eb5bd452a7a45b180f/metricbeat/module/nats/connection/connection_integration_test.go#L40

It might have something to do with the changes in this Dockerfile, does this image get built for you locally?

Reply:
@rdner Well, good catch. I had a local image cache from some version of the Dockerfile before and the integration tests ran locally for me. But now I see what you're seeing after pruning my image cache and rebuilding. I'll get to fixing.

Reply:
@rdner I think I've fixed the test failures. There were a few issues but everything passes locally for me now.

Reply:
@JAndritsch thanks! I'm going to run the full test suite and then ping the codeowners for a review.